foyer/tests/test_plugin.py

```diff
@@ -6,9 +6,9 @@ def test_basic_import():
     assert 'forcefields' in dir(foyer)
 
 
-@pytest.mark.parametrize('ff_name', ['OPLSAA', 'TRAPPE_UA'])
-def test_forcefields_exist(ff_name):
-    ff_name in dir(foyer.forcefields)
+@pytest.mark.parametrize('ff_loader', ['load_OPLSAA', 'load_TRAPPE_UA'])
+def test_forcefields_exist(ff_loader):
+    assert ff_loader in dir(foyer.forcefields)
```

Comment:
What's the purpose of this test? I thought we are trying to move opls and trappe to their own repos, like PFE. 

If we could loop through *all* of the installed plugins somehow, this test would be more all-inclusive and make more sense to have. Try something like the snippet below. Or if there is a cleaner/better way of getting these functions. 

```python3
@pytest.mark.parameterize('ff_loader', 
        [func for func in dir(foyer.forcefields) 
         if 'load' in func and '__' not in func])
```

Reply:
> What's the purpose of this test?

It's just to make sure those loader functions are there, which ...

> I thought we are trying to move opls and trappe to their own repos, like PFE.

assumes, for now that we are shipping these force fields with the main repo.

I like your idea for iterating through the loaders that are in the entry point group gets at what I was going for but isn't this self-assuring? Since it only iterates through the loaders it finds, is there even a way for it to fail? I guess it could fail if the loader is bad, but the point was to check to see the loaders that we hope are there indeed exist.

Reply:
The changes you made now look fine to me, but looking to the future we could probably change a few things: 

At this point the entry point is working on foyer's end. So I think foyer's responsibility should only be to detect whatever plugins in can find in `foyer.forcefields`. The burden is on the maintainers of the specific forcefield plugin repo to make sure their forcefield is detected by the foyer (i.e. they put the correct entry point location or whatever) and that should be checked with their own tests in their own CI. 

So, in my mind, the tests in foyer should only test the forcefields it detects, it shouldn't test if a forcefield is detected correctly.

That being said we could develop a [more sophisticated plugin detector](https://github.com/mosdef-hub/foyer/pull/288#issuecomment-548465083).

Reply:
> The burden is on the maintainers of the specific forcefield plugin repo to make sure their forcefield is detected by the foyer 

That's a good point, I agree